### ISSUE TITLE
Unexpected top-level property "id-match" 

### ISSUE THREAD

**Issue by slim-hmidi** [eslint/eslint#10440]
**Tell us about your environment**

* **ESLint Version:** 4.17.0
* **Node Version:** 8.11.2
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
**babel-eslint:** 8.2.1
**eslint-config-airbnb:** 16.1.0

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 6
    },
    "env": {
        "node": true
    },
    "extends": "airbnb",
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I added `id-match` to the configuration file.

```js
"id-match": ["error", "^[a-z]+([A-Z][a-z]+)*$|^[A-Z]+_{1}[a-z]+([A-Z][a-z]+)*$"]
```

```bash
node_modules/.bin/eslint --ext .jsx,.js ./
```

**What did you expect to happen?**
No error message

**What actually happened? Please include the actual, raw output from ESLint.**
```
ESLint configuration in /home/user1/frontend/.eslintrc.json is invalid:
        - Unexpected top-level property "id-match".

Error: ESLint configuration in /home/user1/frontend/.eslintrc.json is invalid:
        - Unexpected top-level property "id-match".

    at validateConfigSchema /home/user1/frontend/node_modules/eslint/lib/config/config-validator.js:221:15)
    at Object.validate (/home/user1/frontend/node_modules/eslint/lib/config/config-validator.js:238:5)
    at loadFromDisk (/home/user1/frontend/node_modules/eslint/lib/config/config-file.js:516:19)
    at Object.load (/home/user1/frontend/node_modules/eslint/lib/config/config-file.js:559:20)
    at Config.getLocalConfigHierarchy (/home/user1/frontend/node_modules/eslint/lib/config.js:227:44)
    at Config.getConfigHierarchy (/home/user1/frontend/node_modules/eslint/lib/config.js:179:43)
    at Config.getConfigVector (/home/user1/frontend/node_modules/eslint/lib/config.js:286:21)
    at Config.getConfig (/home/user1/frontend/node_modules/eslint/lib/config.js:329:29)
    at CLIEngine.getConfigForFile (/home/user1/frontend/node_modules/eslint/lib/cli-engine.js:653:29)
    at Object.execute (/home/user1/frontend/node_modules/eslint/lib/cli.js:165:39)

**Comment by Aladdin-ADD** [comment]
Thanks for creating the issue. Does adding the config in `rules` work?
```js
{
    // ...
    "extends": "airbnb",
    "rules": {
        "id-match": ["error", "^[a-z]+([A-Z][a-z]+)*$|^[A-Z]+_{1}[a-z]+([A-Z][a-z]+)*$"]
     }
}
```

**Comment by slim-hmidi** [comment]
@Aladdin-ADD thanks it works now.
